### ISSUE TITLE
Multiple stack types

### ISSUE THREAD

**Issue by asardinha** [DimaKudosh/pydfs-lineup-optimizer#139]
Hi Dima,

With the new stacking api, is it possible to add multiple stack types in the optimization. 

For example, if I want to run 150 lineups containing both 4-4 and 4-3 stacks, is it possible? In addition could we make a feature to set exposure for each stack type (such as .60 4-4 and .4 4-3?

Thanks for your help! Much appreciated

**Comment by BenikaH** [comment]
Hi @asardinha. This is can be done with what's currently available. Are you comfortable with JSON format for your parameters?

**Comment by asardinha** [comment]
@BenikaH i am not, but would appreciate the walkthrough. Let me know how you’d like to communicate this method. 

Thanks.

**Comment by Denwen12** [comment]
I'd like to know how I can do this as well

**Comment by brente33** [comment]
Am I missing something on the new Team Stacking? This code doesn't work for me
optimizer.add_stack(TeamStack(3))

I get  optimizer.add_stack(TeamStack(3))
NameError: name 'TeamStack' is not defined

**Comment by austinjohnson** [comment]
Has this been solved yet? The old optimization format used to be able to do multiple stacks. I was trying this morning to generate multiple stack types such as 2-2 or 2-2-2, but it doesn’t seem possible. This also causes an issue because you have to generate lineups that do not know about other lineups therefore your constraints don’t work the way you would expect. @DimaKudosh

**Comment by austinjohnson** [comment]
@BenikaH how did you solve this?

**Comment by BenikaH** [comment]
Got a tad busy, sorry for late reply. Yes, previously in 2.7.0, you could do something like the following.
`stacks = [[2,4],[2,2,2,1]]
 for stack in stacks:
     optimizer.set_team_stacking(stack, for_positions=optimizer.available_positions)
 for lineup in optimizer.optimize(n,randomness=False):
     print(lineup)
`

In this version, I was able to do multiple stacks, but not at the same time. I could do 4-3 and 3-4. I have another idea how to do it. Give me until after lock and I'll play with it a bit more. But if you want to try tonight with a couple of 4-4, 4-3 and so forth, try this.

`for st in team_stacks:
    for ts in st:
        print(ts)
        stack_exposure = ts['stack_exposure']
        optimizer.add_stack(TeamStack(ts['count'],
                                      for_positions=optimizer.available_positions,
                                      for_teams=optimizer.available_teams,
                                      max_exposure=stack_exposure))
            
    for lineup in optimizer.optimize(n=10, randomness=True):
        print(lineup)`

**Comment by BenikaH** [comment]
The code block isn't liking my code lol.

**Comment by austinjohnson** [comment]
Well ideally, you would want to be able to say I want a 2–2 stack for 75 lps and and 2-2-2 stack for the other 75. Out of those two stack types I want every team to have X % exposure. When you generate the stacks outside of each other all of your constraints are lost because the two different generations do not know about one another. Would you agree with my theory? @BenikaH

**Comment by BenikaH** [comment]
Yes I agree with that but it depends on what you're going for. Your constraints aren't lost if you're coding within the scope of your constraints. That is doable.

**Comment by Denwen12** [comment]
I need help with that code for stacking

**Comment by BenikaH** [comment]
@Denwen12 Sure. I was going to jump on this tonight. Got very busy with work.

**Comment by austinjohnson** [comment]
@Denwen12 I would just clone 2.7 and use the stacking method and then if you're generating 150 lps just do 60% (90 lps) 4-4 and the other 40% 4-3 etc...

**Comment by siyu** [comment]
@austinjohnson
I don't see how this can be done in 2.7, below is the 2.7 stacking usage, how can we specify 60% 4-4 and 40% 5-3? 

optimizer.set_team_stacking([3, 3], for_positions=['1B', '2B', '3B', 'C', 'SS', 'OF'])

@DimaKudosh  are you planning to support this feature?

**Comment by lightninglarry** [comment]
anyone know if this feature has been corrected?  with baseball on the way id like to be able to force a 4 man stack and a 3 man stack in every lineup. Is that possible?
